restli-int-test-server/src/main/java/com/linkedin/restli/examples/greetings/server/defaults/FooDefaultFillResources.java

```diff
@@ -0,0 +1,48 @@
+package com.linkedin.restli.examples.greetings.server.defaults;
+
+import com.linkedin.restli.examples.defaults.api.FooDefault;
+import com.linkedin.restli.server.CollectionResult;
+import com.linkedin.restli.server.annotations.Finder;
+import com.linkedin.restli.server.annotations.QueryParam;
+import com.linkedin.restli.server.annotations.RestLiCollection;
+import com.linkedin.restli.server.annotations.RestMethod;
+import com.linkedin.restli.server.resources.CollectionResourceTemplate;
+import java.util.ArrayList;
+import java.util.List;
+
+
+@RestLiCollection(name = "fooDefaultFilling", namespace = "com.linkedin.restli.examples.defaults.api")
+public class FooDefaultFillResources extends CollectionResourceTemplate<Integer, FooDefault>
+{
+  @RestMethod.Get
+  public FooDefault get(Integer keyId)
+  {
+    return new FooDefault().setId(keyId);
+  }
+
+  @RestMethod.GetAll
+  public CollectionResult<FooDefault, FooDefault> getAllFooCollection()
+  {
+    List<FooDefault> fooDefaults = new ArrayList<>();
+    for (int i = 0; i < 10; i++)
+    {
+      fooDefaults.add(new FooDefault().setId(i));
+    }
+    FooDefault customMetadata = new FooDefault().setId(10);
+
+    return new CollectionResult<FooDefault, FooDefault>(fooDefaults, customMetadata);
+  }
+
+  @Finder("findFooCollection")
+  public CollectionResult<FooDefault, FooDefault> findFooCollection(@QueryParam("count") Integer count)
```

Comment:
I understand we have other classes that use the same record for metadata. Since we are testing default filling on metadata, I think using a separate record is cleaner.
For example if default filling is broken for non-entity records, the current setup will not catch it.

Reply:
Yes, In the final test code, I end up using different schema for CollectionResult generics.